learningresources/forms.py

```diff
@@ -35,11 +37,18 @@ class Meta:
     # The ModelForm.save() accepts "commit" and this doesn't, because
     # we always set commit=False then add the user because created_by is
     # not part of the form, and shouldn't be.
+    @transaction.atomic
     def save(self, user):
         """
         Save a newly-created form.
         """
         repo = super(RepositoryForm, self).save(commit=False)
+        slug = slugify(repo.name)
+        count = 1
+        while Repository.objects.filter(slug=slug).exists():
```

Comment:
I don't think so. I did some looking and although Django does have a [SlugField](https://docs.djangoproject.com/en/1.8/ref/models/fields/#slugfield), [this post](https://keyerror.com/blog/automatically-generating-unique-slugs-in-django) seems to indicate that Django won't automatically prevent dupes even if you set `unique=True`. If you find something that lets me eliminate this loop please let me know, though. That would be nice.